File: app/src/main/java/mchehab/com/knearestneighbor/distanceAlgorithm/EuclideanDistance.java
package mchehab.com.knearestneighbor.distanceAlgorithm;

public class EuclideanDistance implements DistanceAlgorithm {
    @Override
    public double calculateDistance(double x1, double y1, double x2, double y2) {
        double xSquare = Math.pow(x1 - x2, 2);
        double ySquare = Math.pow(y1 - y2, 2);
        double distance = Math.sqrt(xSquare + ySquare);
        return distance;
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/distanceAlgorithm/EuclideanDistance.kt
package mchehab.com.kotlin.distanceAlgorithm

import kotlin.math.pow
import kotlin.math.sqrt

class EuclideanDistance: DistanceAlgorithm{
    override fun calculateDistance(x1: Double, x2: Double, y1: Double, y2: Double): Double{
        val xSquare = (x1 - x2).pow(2)
        val ySquare = (y1 - y2).pow(2)
        val distance = sqrt(xSquare + ySquare)
        return distance
    }
}